test/ViewsTest.cpp

```diff
@@ -272,3 +275,79 @@ TEST(Views, CallbackOnEndView) {
   // Callback not invoked for the destructor of the moved-from `viewA`.
   EXPECT_EQ(numCalls, 3);
 }
+// _____________________________________________________________________________
+TEST(Views, RvalueView) {
+  // Initial value is `true` and when being moved from it will be `false`.
+  using MoveTracker = ad_utility::ResetWhenMoved<bool, false, true>;
+  // not moved from.
+  const auto t = MoveTracker{true};
+  // moved from.
+  const auto f = MoveTracker{false};
+
+  // This impl tests the different ways an `RvalueView` can be created:
+  // Either from a const or mutable input (first argument of type
+  // `ValueIdentity<bool>`, And the view is either copied or moved into the
+  // place where it's used (second argument).
+  auto testImpl = [&t, &f](auto isConst, bool doMove,
+                           ad_utility::source_location loc =
+                               ad_utility::source_location::current()) {
+    auto tr = generateLocationTrace(loc);
+    std::vector<MoveTracker> vec;
+    vec.resize(10);
+
+    std::vector<MoveTracker> target;
```

Comment:
Ah that's where it's being used. Yeah you don't need the additional template argument if you just initialise `vec` differently.